src/sdk/cloudformation.ts

```diff
@@ -0,0 +1,131 @@
+import AWS, { CloudFormation } from 'aws-sdk'
+
+import config from 'config'
+import { getStackName } from '../cdk/app'
+import { getRoleArn, assumeRole } from '../assume-role'
+import { MiraConfig } from '../config/mira-config'
+
+/**
+ * Gets the full stack name.
+ * @todo Refactor this into a more proper place.
+ */
+export const getFullStackName = (): string => {
+  return `${getStackName()}-${MiraConfig.getEnvironment().name}`
+}
+
+let miraCfn: AWS.CloudFormation
+/**
+ * Gets a Cfn object.
+ */
+export const getCfn = async (): Promise<CloudFormation> => {
+  if (miraCfn) {
+    return miraCfn
+  }
+  const awsConfig = await assumeRole(getRoleArn(config.get(`accounts.${MiraConfig.getEnvironment().name}.profile`)))
+  awsConfig.region = MiraConfig.getEnvironment().env.region
+  AWS.config = awsConfig
+  miraCfn = new AWS.CloudFormation({ apiVersion: '2010-05-15' })
+  return miraCfn
+}
+
+/**
+ * Gets the stacks that are in progress.  This can be used to detect if a stack
+ * is in progress of being pushed.
+ * @todo
+ */
+// export const getInProgressStacks = (): void => {
+
+// }
+
+/**
+ * Gets the resources included in a stack.
+ */
+export const getStackResources = async (StackName: string): Promise<LooseObject> => {
+  const cfn = await getCfn()
+  return new Promise((resolve, reject) => {
```

Comment:
All the SDK method can used directly using the promise adding `.promise()`  and skip the callback param

(EG.)
```
      listResult = await s3
        .listObjectsV2({
          Bucket: config.trail.bucketName,
          Prefix: `${prefix}/${config.region}/${day}`,
          ContinuationToken: listResult
            ? listResult.NextContinuationToken
            : undefined
        })
        .promise()
```

Reply:
Done